drivers/atrvt100.s

```diff
@@ -1,6 +1,1709 @@
+
+ErBeLn  ldx ROWCRS
+        jsr SLV
+        ldy COLCRS
+        lda #$00
+EBL1    sta (xVector),y
+        dey
+        bpl EBL1
+        sta sCrsrChar ; del char ..
+                      ; ..under crsr
+        jsr COn
+        rts
+
+; -------------------------------------
+; SLV - set line vector
+; --- INTERNAL ---
+;
+; params: line no in X
+; affects: A, Y
+; return: screen line ptr in xVector
+; -------------------------------------
+
+SLV     lda     SAVMSC
+        clc
+        adc     LineOffsLo,x
+        sta     xVector
+        lda     SAVMSC+1
+        adc     LineOffsHi,x
+        sta     xVector+1
+        rts
+
+LineOffsLo
+        .repeat Rows, i
+        .byt    (Cols * i) & $ff
+        .endrepeat
+LineOffsHi
+        .repeat Rows, i
+        .byt    (Cols * i) >> 8
+        .endrepeat
+
+; *************************************
+; *
+; * Init routines
+; *
+; *************************************
+
+; -------------------------------------
+; init memory
+;
+; -------------------------------------
+
+InitVar lda #$00
+        sta EMode
+        sta EBufL
+        sta SRS
+        sta Font
+        sta Draw
+        sta SaveF
+        sta SaveR
+        sta SaveD
+        sta SaveRow
+        sta SaveCol
+        sta lbPending
+        sta civis
+        sta Rvs
+
+        lda #Rows-1 ; last line
+        sta SRE     ; = Rows-1
+
+        rts
+
+; -------------------------------------
+; Init char
+;
+; -------------------------------------
+
+InitChar
+        rts
+
+; -------------------------------------
+; Exit char
+;
+; -------------------------------------
+
+ExitChar
+        rts
+
+; -------------------------------------
+; Init Screen
+;
+; -------------------------------------
+
+InitScr
+        lda LMARGN
+        sta LMARGN_save
+        lda SHFLOK
+        sta SHFLOK_save
+        lda INVFLG
+        sta INVFLG_save
+        lda #0
+        sta LMARGN
+        sta SHFLOK
+        sta INVFLG
+InitScr1              ; called from ExitScr
+        ldx #Rows-1
+IS1     txa
+        pha           ; save X
+        jsr ErLn      ; erase line
+        pla
+        tax           ; restore X
+        dex           ; previous line
+        bpl IS1
+        ; --- put crsr ---
+        ldx #$00
+        ldy #$00
+        sty COLCRS+1  ; sanity
+        jsr Plot
+
+        rts
+
+; -------------------------------------
+; Exit Screen
+;
+; -------------------------------------
+
+ExitScr
+        lda LMARGN_save
+        sta LMARGN
+        lda SHFLOK_save
+        sta SHFLOK
+        lda INVFLG_save
+        sta INVFLG
+        jmp InitScr1
+
+; *************************************
+; *
+; * ASCII and ScreenCode tables
+; *
+; *************************************
+
+; -------------------------------------
+; table line drawing to ScreenCode
+;
+; This tabel is used to convert incoming
+; line drawing chars.
+; -------------------------------------
+
+ltsc;_0  _1  _2  _3  _4  _5  _6  _7  _8  _9  _a  _b  _c  _d  _e  _f
```

Comment:
Adjusted, but incompletely. A better (and shorter) table will come with the font change.